source/validate_instruction.cpp

```diff
@@ -131,6 +131,68 @@ spv_result_t CapCheck(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Checks that the number of characters in a string doesn't exceed the limit.
+spv_result_t LimitCheckString(ValidationState_t& _,
+                              const spv_parsed_instruction_t* inst) {
+  // By definition, num_words of an operand is a 16-bit unsigned which will not
+  // exceed the limit of characters for a string literal.
+  // This function checks that a NULL termination must be found before reaching
+  // the end of the string words.
+  for (int i = 0; i < inst->num_operands; ++i) {
+    auto operand = inst->operands[i];
+    if (SPV_OPERAND_TYPE_LITERAL_STRING == operand.type) {
+      bool null_terminated = false;
```

Comment:
Dang.  I found it in http://en.cppreference.com/w/cpp/string/byte/strlen  
But that says "Since C11". And that's not C++11.  Ugh.

Alternatives are std::find or std::memchr.  I think std::memchr is the better choice.
```
const size_t width_in_bytes = operand.num_words * sizeof(uint32_t);
const bool null_terminated = nullptr != std::memchr(reinterpret_cast<const char*>(inst->words + operand.offset), 0, width_in_bytes);
```